go.mod

```diff
@@ -1,5 +1,9 @@
 module github.com/aws/aws-secretsmanager-caching-go
 
-go 1.12
+go 1.15
```

Comment:
Thanks for this PR!
Wonder if its worth bumping to go 1.19 or 1.20?

Reply:
The go.mod for aws-sdk-go-v2 says go 1.15
Depends on how you see the go.mod version. I think it says "the min version you need to run this.
So I would vote for 1.15 and increase it as soon newer features like generics are used.